lib/std/os.zig

```diff
@@ -210,6 +219,13 @@ pub const LOG = struct {
     pub const DEBUG = 7;
 };
 
+pub const RelativePath = struct {
```

Comment:
This looks an WASI-only abstraction; should it not go into `wasi.zig` submodule? Perhaps together with preopens machinery?

Reply:
If I understood @andrewrk correctly on Discord, it sounds like only purely native functionality should be available in `std/os/wasi.zig`. This means that platform-specific code for implementing POSIX support has to stay in std/os.zig, just along with the existing platform-specific POSIX logic (e.g. openatWasi and openOptionsFromFlagsWindows)

Ideally, `std/os/wasi.zig` would exist for the purely native stuff, and then `std/posix/wasi.zig` could be a home for the WASI-specific implementation of POSIX-like behavior, but without #5019 we don't have that separation

Let me know if I got any of that wrong - Happy to fix it up if there's a mistake here :+1:

Reply:
Mhm, that's fair. But given that it's only used by WASI target, perhaps a more meaningful name such as `RelativePathWasi` or `WasiRelativePath` would make sense? Also, if you could drop a general doc-style comment for the struct - its intended use, etc. - that would great!

Reply:
Sure thing!

TBH, I think the clarity of the `*at` functions would be improved if they accepted a `RelativePath` rather than separate fd/path parameters related by name (`symlinkat` is particularly confusing). That's a separate change though and perhaps too far away from the standard POSIX interface, so I'll just document this as WASI-specific for now.

Reply:
Actually, that's a neat idea. Would you be willing to write it up as a proposal issue?

Reply:
Absolutely - I'll file something this evening when I come around to this PR again